File: les2/scripts/combine_counts.R
lines <- read.table(snakemake@input[["lines"]], header=FALSE)
words <- read.table(snakemake@input[["words"]], header=FALSE)

num_lines <- lines[1]
num_words <- words[1]

out_line <- paste("lines", "\t", num_lines, "\n", "words", "\t", num_words)
writeLines(out_line, snakemake@output[[1]])

File: les2/scripts/combine_counts.py
with open(snakemake.input.lines) as line_reader, 
    open(snakemake.input.words) as word_reader, 
    open(snakemake.output[0], "w") as out_writer:

    num_lines = line_reader.read().split()[0]
    num_words = word_reader.read().split()[0]

    out_writer.write(f"lines\t{num_lines}\nwords\t{num_words}\n")